static/js/message_list_view.js

```diff
@@ -132,6 +132,59 @@ function update_message_date_divider(opts) {
         timerender.render_date(curr_time, prev_time, today)[0].outerHTML;
 }
 
+function clear_unread_messages_divider(msg) {
+    msg.want_unread_divider = false;
+    msg.unread_divider_html = undefined;
+}
+
+function add_unread_messages_divider(msg) {
+    msg.want_unread_divider = true;
+    msg.include_sender = true;
+
+    var NEW_MESSAGES_TEXT = 'NEW MESSAGES';
+
+    if (msg.want_date_divider) {
+        msg.want_date_divider = false;
```

Comment:
I'm a little skeptical of settings `want_date_divider` to false; do we need to do that?  I could imagine some future code trying to check if there's a date divider being confused by this.

Reply:
Yeah I removed it. The `want_unread_divider` takes preference in the template so it won't be a problem.